OpenRA.Mods.Common/Traits/World/PathFinder.cs

```diff
@@ -22,66 +21,85 @@ namespace OpenRA.Mods.Common.Traits
 	[Desc("Calculates routes for mobile units based on the A* search algorithm.", " Attach this to the world actor.")]
 	public class PathFinderInfo : ITraitInfo
 	{
-		public object Create(ActorInitializer init) { return new PathFinder(init.World); }
+		public object Create(ActorInitializer init) { return new PathFinder(init.World, new NonCachedPathFinder(init.World)); }
 	}
 
-	public class PathFinder
+	public interface IPathFinder
 	{
-		const int MaxPathAge = 50;	/* x 40ms ticks */
-		static readonly List<CPos> EmptyPath = new List<CPos>(0);
+		List<CPos> FindUnitPath(CPos from, CPos target, Actor self);
+		List<CPos> FindUnitPathToRange(CPos src, SubCell srcSub, WPos target, WRange range, Actor self);
+		List<CPos> FindPath(PathSearch search);
+		List<CPos> FindBidiPath(PathSearch fromSrc, PathSearch fromDest);
+	}
 
-		readonly World world;
-		public PathFinder(World world) { this.world = world; }
+	public class PathFinder : IPathFinder
+	{
+		const int MaxPathAge = 50;	/* x 40ms ticks */
 
 		class CachedPath
```

Comment:
It was like that when I took the file, I didn't feel like chaging it right now, but I agree to put constants elsewhere.